src/sqlfluff/rules/L039.py

```diff
@@ -75,18 +77,14 @@ def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
                 # https://github.com/sqlfluff/sqlfluff/issues/1304
                 # It represents the question: are we parsing through leading whitespace
                 # in this loop?
-                leading_whitespace = True
                 for child_seg in seg.get_raw_segments():
                     if child_seg.is_whitespace:
-                        if not leading_whitespace:
-                            violations.append(
-                                LintResult(
-                                    anchor=child_seg,
-                                    fixes=[LintFix.delete(child_seg)],
-                                )
+                        violations.append(
+                            LintResult(
+                                anchor=child_seg,
+                                fixes=[LintFix.delete(child_seg)],
                             )
-                    else:
-                        leading_whitespace = False
+                        )
```

Comment:
The `leading_whitespace` logic was a workaround for issue #1304. This PR fixes that issue, so this code is no longer needed.